copier/config/user_data.py

```diff
@@ -108,8 +112,36 @@ def cast_answer_type(answer: Any, type_fn: Callable) -> Any:
         return answer
 
 
+def render_string(template, env, context):
+    """Render a templated-string using Jinja."""
+    try:
+        return env.from_string(template).render(**context)
+    except TypeError:
+        # template was not a string
+        return template
+    except UndefinedError as error:
+        raise UserMessageError(str(error)) from error
+
+
+def render_choices(choices, env, context):
+    """Render a list of templated choices using Jinja."""
+    render = partial(render_string, env=env, context=context)
+    if isinstance(choices, dict):
+        choices = {render(k): render(v) for k, v in choices.items()}
+    elif isinstance(choices, list):
+        for i, choice in enumerate(choices):
+            if isinstance(choice, (tuple, list)) and len(choice) == 2:
+                choices[i] = (render(choice[0]), render(choice[1]))
+            else:
+                choices[i] = render(choice)
+    return choices
+
+
 def query_user_data(
-    questions_data: AnyByStrDict, answers_data: AnyByStrDict, ask_user: bool
+    questions_data: AnyByStrDict,
+    answers_data: AnyByStrDict,
+    ask_user: bool,
+    envops: dict,
```

Comment:
Following one comment above:

```suggestion
    envops: EnvOps,
```

Reply:
Done.